### PR TITLE
feat(backend): subscription to calls you are invited

### LINKED ISSUE #2229: 🚀 [Feature] Subscribe to Calls you are invited to
## 🚀 Feature

Subscribe to Calls you are invited to

This feature applies only to Malltalk tables

Requires https://github.com/dreammall-earth/dreammall.earth/issues/2228 for a full implementation

The Idea is to show the user a Call Notification he can accept or decline when he is invited to a Table.
image

The Task of this Ticket is:
- Have the frontend subscribe to call you receive
- Have the backend fire calls whenever a Malltalk Table is opened (public or private) on that subscription interface
- The frontend shall log that a call is received

This Ticket does not include the Frontend visual Implementation. See here: https://github.com/dreammall-earth/dreammall.earth/issues/2231

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest

Motivation
---
As user I want to be informed that I am invited to a call so that I can join it if I am present

Solution
---
Create a subscription that returns a InvitedTable object so it can be shown in the frontend

### Issues

- fixes #2229

### Todo

- [X] None

### CODE REVIEW

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolvers/TableResolver.ts [hunk 1]
```diff
@@ -75,6 +77,20 @@ export class TableResolver {
...
+        table: {
+          id: String(meeting.id),
+          isModerator: false,
+          meetingID: meeting.meetingID,
+          meetingName: meeting.name,
+          startTime: String(meeting.createTime),
+          attendees: [],
+          participantCount: 0,
+        },
+      })
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolvers/TableResolver.ts:93]: You do not test this to be called

File: backend/src/graphql/resolvers/TableResolver.ts [hunk 2]
```diff
@@ -528,6 +543,26 @@ export class TableResolver {
...
+      if (payload.userIds.some((id: any) => currentUser.id === id)) {
+        return true // Allow this subscription for the recipient
+      }
+
+      return false // Deny this subscription
+    },
+  })
+  call(@Root() call: Call): Call {
+    return call
+  }
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolvers/TableResolver.ts:564]: This code is untested

File: frontend/src/graphql/subscriptions/callSubscription.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { gql } from 'graphql-tag'
+
+export const callSubscription = gql`
+  subscription {
+    inviteCall {
```
Comment (ulfgebhardt) [inline frontend/src/graphql/subscriptions/callSubscription.ts:5]: I do not understand how the subscription is properly assigned to `inviteCall` - how does this correlate to `'CALL_SUBSCRIPTION'` defined above?

File: backend/src/graphql/resolvers/TableResolver.ts [hunk 4]
```diff
@@ -545,6 +559,45 @@ export class TableResolver {
...
+  })
+  async inviteTable(
+    @Root() invitedTable: InvitedTable,
+    @Ctx() context: Context,
+  ): Promise<InvitedTable | null> {
+    const {
+      user,
+      dataSources: { prisma },
+    } = context
+    if (!user) return null
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply backend/src/graphql/resolvers/TableResolver.ts]: Right

**Review by ulfgebhardt — APPROVED** [review]
Lets merge this and solve the test problem in another part: https://github.com/dreammall-earth/dreammall.earth/issues/2642

**Pull request merged**